src/http_server.cc

```diff
@@ -3060,120 +3104,449 @@ HTTPAPIServer::HandleInfer(
 
   // HTTP request paused when creating inference request. Resume it on exit if
   // this function returns early due to error. Otherwise resumed in callback.
-  bool connection_paused = true;
-  auto infer_request = CreateInferRequest(req);
-  infer_request->trace_ = trace;
+  std::unique_ptr<GenerateRequestClass> generate_request;
+  if (streaming) {
+    generate_request.reset(new GenerateRequestClass(
+        server_.get(), req, GetResponseCompressionType(req),
+        generate_stream_request_schema_.get(),
+        generate_stream_response_schema_.get(), streaming, irequest));
+  } else {
+    generate_request.reset(new GenerateRequestClass(
+        server_.get(), req, GetResponseCompressionType(req),
```

Comment:
So, we are supporting the response compression but not request compression? We should include it in our testing if we are planning to leave it here.

Reply:
No, both directions are not supported, this is passed due to the parent class expects it in constructor. The compression logic is not properly encapsulated so it doesn't automatically kick in, and frankly I think some investigation is needed to figure out how chunk + compression works. Therefore, compression is not under consideration for now.

Reply:
Instead of getting it from the req - can we error if it's not IDENTITY and/or pass IDENTITY as the type (and leave a note / TODO)

Reply:
@GuanLuo - recommend we put the same error handling as in line 3090 - can we add check for responsecompressiontype there as well - to ensure it is IDENTITY?